src/Tasks/Common/Resources/Strings.resx

```diff
@@ -588,4 +588,47 @@ The following are names of parameters or literal values and should not be transl
     <value>NETSDK1120: C++/CLI projects targeting .NET Core require a target framework of at least 'netcoreapp3.1'.</value>
     <comment>{StrBegin="NETSDK1120: "}</comment>
   </data>
+  <data name="AppHostMachONotExpectedFormat" xml:space="preserve">
+    <value>NETSDK1121: Unable to use '{0}' as macOS application host executable it was not in the expected format. Detail reason: {1}.</value>
+    <comment>{StrBegin="NETSDK1121: "}</comment>
+  </data>
+  <data name="AppHostMachOFormatNot64BitExe" xml:space="preserve">
```

Comment:
Do we want to print the details about what went wrong? It seems like more to do with the implementation of the SDK/HostModel. Any failure should be straightforward to repro on our end.  

I'm happy that you've already done the work ... so might as well keep it.